qcodes_contrib_drivers/drivers/Attocube/ANC350Lib/interface.py

```diff
@@ -0,0 +1,158 @@
+"""Collection of shared classes used by multiple versions of ANC350Lib
+
+Author:
+    Lukas Lankes, Forschungszentrum Jülich GmbH / ZEA-2, l.lankes@fz-juelich.de
+"""
+
+from abc import ABC, abstractmethod
+from enum import IntEnum
+from typing import List, Optional
+import warnings
+
+__all__ = ["ANC350LibError", "ANC350LibDeviceType", "ANC350LibExternalTriggerMode",
+           "ANC350LibTriggerPolarity", "ANC350LibActuatorType", "ANC350LibAmplitudeControlMode",
+           "ANC350LibSignalEdge", "ANC350LibTriggerInputMode", "ANC350LibTriggerOutputMode"]
+
+
+class ANC350LibError(Exception, ABC):
+    """Base class for exceptions occurring in ANC350v?Lib-classes
+
+    Attributes:
+        message: Error message
+        code: Error code from dll (or None)
+    """
+    SUCCESS_CODES: List[int] = [0]
+    WARNING_CODES: List[int] = []
+
+    def __init__(self, message: Optional[str], code: Optional[int]):
+        """Create instance of ``ANC350LibError``
+
+        Args:
+            message: Error message
+            code: Error code from dll
+        """
+        self.code = code
+        self.message = self._process_code_message(code, message)
+
+        super().__init__(self.message)
+
+    @classmethod
+    def check_error(cls, code: int, function: Optional[str] = None) -> None:
+        """Checks the error code and raises an exception, if necessary
+
+        If the error code is in ``SUCCESS_CODES``, no exception is raises. Same applies if the code
+        is contained in ``WARNING_CODES``. Then, a warning is generated. If neither of both is the
+        case, an exception is generated based on the error code.
+
+        Args:
+            code: Occurred error code
+            function: Dll-function that returned the error code
+
+        Raises:
+            If code is not in ``SUCCESS_CODES`` or ``WARNING_CODES``, an ANC350LibError is raised
+        """
+        if code not in cls.SUCCESS_CODES:
+            if function:
+                function = "In function " + function
+            if code in cls.WARNING_CODES:
+                warnings.warn(cls._process_code_message(code, function))
+            else:
+                raise cls(function, code)
+
+    @classmethod
+    def _process_code_message(cls, code: Optional[int], message: Optional[str]) -> str:
+        """Converts error code and message into a more detailed error message
+
+        Args:
+            code: Occurred error code
+            message: Error message
+
+        Returns:
+            Detailed error message
+        """
+        message_out = "Error in ANC350Lib"
+        if code is not None:
+            message_out += " ({}): ".format(code)
```

Comment:
Since you changed string format as f"" everywhere, you may change it here

Reply:
Oh, sorry I forgot to check the other files

Reply:
No worries.